cmd/minishift/cmd/root.go

```diff
@@ -254,3 +282,39 @@ func performPostUpdateExecution(markerPath string) error {
 
 	return nil
 }
+
+func ensureAllInstanceConfigPath(configPath string) {
+	configDir := filepath.Dir(configPath)
+	if err := os.MkdirAll(configDir, 0777); err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating directory: %s", configDir))
+	}
+}
+
+func createDirForMinishiftRun(dirs *minishiftConfig.MinishiftDirs) {
+	dirPaths := reflect.ValueOf(*dirs)
+
+	for i := 0; i < dirPaths.NumField(); i++ {
+		path := dirPaths.Field(i).Interface().(string)
+		if err := os.MkdirAll(path, 0777); err != nil {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error creating directory: %s", path))
+		}
+	}
+}
+
+//If there is no active profiles we need to set
+// minishift as the default profile if minishift profile exists
+func setDefaultActiveProfile() {
+	if minishiftConfig.AllInstancesConfig == nil {
+		atexit.ExitWithMessage(1, "Error: All instance config is not initialized")
+	}
+	activeProfile := profileActions.GetActiveProfile()
+	if activeProfile == "" {
+		err := profileActions.SetActiveProfile(constants.DefaultProfileName)
+		if err != nil {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error while setting default profile '%s': %s", profile, err.Error()))
+		}
+		if constants.ProfileName == constants.DefaultProfileName {
```

Comment:
does it matter to check if profile here is the default?

I think this is only relevant to be checked in `cmdUtil.setOcContext` itself... but correct me if wrong:

```diff
-		if constants.ProfileName == constants.DefaultProfileName {
-			cmdUtil.SetOcContext(constants.DefaultProfileName)
-		}
+		cmdUtil.SetOcContext(constants.ProfileName)
```
the same.... isn't the whole point of this function not to set the default profile active?